pkg/cli/options_test.go

```diff
@@ -34,6 +35,167 @@ import (
 )
 
 var _ = Describe("Discover external plugins", func() {
+	Context("with valid plugins root path", func() {
+		var (
+			homePath           string = os.Getenv("HOME")
+			customPath         string = "/tmp/myplugins"
+			xdghome            string = os.Getenv("XDG_CONFIG_HOME")
+			originalPluginPath string = os.Getenv("KB_PLUGINS_PATH")
+		)
+
+		AfterEach(func() {
+			if originalPluginPath != "" {
+				// restore the original value
+				err := os.Setenv("KB_PLUGINS_PATH", originalPluginPath)
+				Expect(err).To(BeNil())
+			} else {
+				// unset if it was originally unset
+				err := os.Unsetenv("KB_PLUGINS_PATH")
+				Expect(err).To(BeNil())
+			}
+		})
+
+		When("XDG_CONFIG_HOME is not set and using the $HOME environment variable", func() {
+			BeforeEach(func() {
+				err := os.Unsetenv("KB_PLUGINS_PATH")
+				Expect(err).To(BeNil())
+
+				err = os.Unsetenv("XDG_CONFIG_HOME")
+				Expect(err).To(BeNil())
+
+				xdghome = os.Getenv("XDG_CONFIG_HOME")
+			})
+
+			It("should return the correct path for the darwin OS", func() {
+				plgPath, err := getPluginsRoot("darwin")
+				Expect(err).To(BeNil())
+				Expect(plgPath).To(Equal(fmt.Sprintf("%s/Library/Application Support/kubebuilder/plugins", homePath)))
+			})
+
+			It("should return the correct path for the linux OS", func() {
+				plgPath, err := getPluginsRoot("linux")
+				Expect(err).To(BeNil())
+				Expect(plgPath).To(Equal(fmt.Sprintf("%s/.config/kubebuilder/plugins", homePath)))
+			})
+
+			It("should return error when the host is not darwin / linux", func() {
+				plgPath, err := getPluginsRoot("random")
+				Expect(plgPath).To(Equal(""))
+				Expect(err).ToNot(BeNil())
+				Expect(err.Error()).To(ContainSubstring("host not supported"))
+			})
+		})
+
+		When("XDG_CONFIG_HOME is set", func() {
+			BeforeEach(func() {
+				err := os.Unsetenv("KB_PLUGINS_PATH")
+				Expect(err).To(BeNil())
+
+				err = os.Setenv("XDG_CONFIG_HOME", fmt.Sprintf("%s/.config", homePath))
+				Expect(err).To(BeNil())
+
+				xdghome = os.Getenv("XDG_CONFIG_HOME")
+			})
+
+			AfterEach(func() {
+				err := os.Unsetenv("XDG_CONFIG_HOME")
+				Expect(err).To(BeNil())
+			})
```

Comment:
With this before/after each set, what happens if I set my `XDG_CONFIG_HOME` environment variable and then run this test in the same session?

IIUC this would overwrite my current value with the test value and then completely unset it, leaving my `XDG_CONFIG_HOME` variable different from what I set it to before running the tests.